packages/facility-server/app/routes/apiv1/patient/patientprogramRegistration/patientProgramRegistration.js

```diff
@@ -111,6 +93,59 @@ patientProgramRegistration.post(
   }),
 );
 
+patientProgramRegistration.put(
+  '/programRegistration/:programRegistrationId',
+  asyncHandler(async (req, res) => {
+    req.checkPermission('write', 'PatientProgramRegistration');
+    const { db, models, params, body } = req;
+    const { programRegistrationId } = params;
+    const { conditions = [], ...registrationData } = body;
+
+    if (conditions.length > 0) {
+      req.checkPermission('create', 'PatientProgramRegistrationCondition');
+    }
+
+    const existingRegistration =
+      await models.PatientProgramRegistration.findByPk(programRegistrationId);
+
+    if (!existingRegistration) {
+      throw new NotFoundError('PatientProgramRegistration not found');
+    }
+
+    const { patientId, programRegistryId } = existingRegistration;
+
+    const conditionsData = conditions
+      .filter((condition) => condition.id)
```

Comment:
Out curiosity, when would we be passing in conditions without an `id`?

Reply:
There isn't actually a valid scenario when that should happen. My intention was validation so I've updated this to throw an error if there are any conditions without an id